src/models/app_data_json_format.rs

```diff
@@ -175,88 +174,69 @@ mod tests {
-                    "buyAmount": "123456789",
-                    "sellAmount": "123456789101112",
-                }
-            }
-        });
-        let json: AppData = serde_json::from_value(value).unwrap();
-        let expected = AppData {
-            version: "1.2.3".to_string(),
-            app_code: "MooSwap".to_string(),
-            environment: Some("production".to_string()),
-            metadata: Some(Metadata {
-                environment: Some("production".to_string()),
-                referrer: Some(Referrer {
-                    address: "0x8c35B7eE520277D14af5F6098835A584C337311b"
-                        .parse()
-                        .unwrap(),
-                    version: "6.6.6".to_string(),
-                }),
-                // Notice that although we have specified v2 for quote, we fall back on the v1 quote.
-                // This is much cleaner than trying to implement optional fields for each of
-                // the three components and validate that
-                // only one of slippage OR (both of sell/buy amounts are populated)
-                quote: Some(Quote::V1(QuoteV1 {
-                    version: String::from("2.0"),
-                    buy_amount: U256::from_dec_str("123456789").unwrap(),
-                    sell_amount: U256::from_dec_str("123456789101112").unwrap(),
-                })),
+                quote: Some(Quote::V2(QuoteV2 { slippage_bips: 5 })),
             }),
         };
 
         assert_eq!(json, expected);
     }
 
     #[test]
-    fn test_loading_quote() {
+    fn test_loading_quote_ok() {
         let v1_value = json!({
-            "version": "2.0",
+            "version": "0.1.0",
             "sellAmount": "123",
             "buyAmount": "4567",
         });
         let v2_value = json!({
-            "version": "2.0",
+            "version": "0.2.0",
             "slippageBips": "100"
         });
         let json_1: Quote = serde_json::from_value(v1_value).unwrap();
         let json_2: Quote = serde_json::from_value(v2_value).unwrap();
         let expected_1 = Quote::V1(QuoteV1 {
-            version: String::from("2.0"),
             sell_amount: U256::from_dec_str("123").unwrap(),
             buy_amount: U256::from_dec_str("4567").unwrap(),
         });
 
-        let expected_2 = Quote::V2(QuoteV2 {
-            version: String::from("2.0"),
-            slippage_bips: 100,
-        });
+        let expected_2 = Quote::V2(QuoteV2 { slippage_bips: 100 });
 
         assert_eq!(json_1, expected_1);
         assert_eq!(json_2, expected_2);
     }
+
+    #[test]
+    fn test_loading_quote_err() {
+        // V2 version with V1 data
+        assert_eq!(
+            serde_json::from_value::<Quote>(json!({
+                "version": "0.2.0",
+                "sellAmount": "123",
+                "buyAmount": "4567",
+            }))
+            .unwrap_err()
+            .to_string(),
+            "missing field `slippageBips`"
```

Comment:
nit: I find tests that assert error messages brittle since they depend on implementation details of external crates. Here, `serde_derive` crate is generating these messages, and if they decide to change the error message for `"found missing filed ..."` then the test will start failing. I think `assert!(....is_err())` should be enough.

Reply:
Note that I am OK with how it just, just a small "nit" suggestion.

Reply:
Totally legit reason. Will fix.